plugin/keyring_vault/i_vault_curl.h

```diff
@@ -0,0 +1,23 @@
+#ifndef MYSQL_I_VAULT_CURL
+#define MYSQL_I_VAULT_CURL
+
+#include "i_keyring_key.h"
+#include "vault_credentials.h"
+
+namespace keyring {
+
+class IVault_curl : public Keyring_alloc
+{
+public:
+  virtual my_bool init(Vault_credentials *vault_credentials) = 0;
+  virtual my_bool list_keys(std::string *response) = 0;
+  virtual my_bool write_key(IKey *key, std::string *response) = 0;
```

Comment:
key cannot be const as I am using key's method which are not marked as const in IKey interface. Also not non-const references cannot be used as methods arguments.